editor/src/clj/editor/code/view.clj

```diff
@@ -3544,11 +3494,15 @@
 
 (defn- make-view! [graph parent resource-node opts]
   (let [{:keys [^Tab tab app-view grammar open-resource-fn project]} opts
+        basis (g/now)
+        resource-node-type (g/node-type* basis resource-node)
+        editable (and (g/has-property? resource-node-type :modified-lines)
+                      (g/connected? basis resource-node :save-data project :save-data))
         grid (GridPane.)
         canvas (Canvas.)
         canvas-pane (Pane. (into-array Node [canvas]))
         undo-grouping-info (pair :navigation (gensym))
-        lsp (lsp/get-node-lsp resource-node)
+        lsp (lsp/get-node-lsp basis resource-node)
```

Comment:
Tiny cleanup: We can reuse the `basis` from above to evaluate this as well. If we do not supply it here, `get-node-lsp` will implicitly call `(g/now)` internally.